src/Drawer/ImplicitRouteBinding.php

```diff
@@ -104,20 +104,16 @@ protected function resolveParameter($route, $parameterName, $parameterClassName)
         $parent = $route->parentOfParameter($parameterName);
 
         if ($parent instanceof UrlRoutable && array_key_exists($parameterName, $route->bindingFields())) {
```

Comment:
This nested `if {} else { if {} else {} }` looks like something that could be written really neatly with `match(true)`. Or at least I'd change it from a nested if to `if {} elseif {} else {}`.

Reply:
@levu42 It was an if elseif elseif and it was harder to read. I tried a match but due to the different conditions this seemed the simplest way to write it and still have readable code. Open to suggestions though if you can share a nice looking code snippet 🙂

Reply:
@joshhanley I was thinking of this:

```php
        $model = match(true) {
            $parent instanceof UrlRoutable && array_key_exists($parameterName, $route->bindingFields()) =>
                $parent->resolveChildRouteBinding($parameterName, $parameterValue, $route->bindingFieldFor($parameterName)),

            $route->allowsTrashedBindings() =>
                $instance->resolveSoftDeletableRouteBinding($parameterValue, $route->bindingFieldFor($parameterName)),

            default => $instance->resolveRouteBinding($parameterValue, $route->bindingFieldFor($parameterName))
        };
```

But it's also not the greatest in terms of readability.